source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+   cd /test_repo
+   git clone https://github.com/NVIDIA/cuda-samples.git
+   cd /test_repo/cuda-samples/Samples/2_Concepts_and_Techniques/eigenvalues/
+   make
+
+   %runscript
+   #Executes when the "singularity run" command is used
+   #Useful when you want the container to run as an executable
+   cd /test_repo/cuda-samples/Samples/2_Concepts_and_Techniques/eigenvalues/
+   ./eigenvalues
+
+   %help
+   This is a demo container to show how to build and run a CUDA application
+   on a GPU node
+
+This container will take a base image from `docker-hub <https://hub.docker.com/>`_ and use pre-built `nvidia/cuda <https://hub.docker.com/r/nvidia/cuda>`_ container of a specific version. This container also contains the necessary CUDA tools to compile binaries that run on GPUs. After starting from this base-image, in the next steps some tools are installed, directories are created and filled with a git repository. From this repository a single example of a CUDA applictation is compiled. When running the container on the command line, this application is run automatically.
+
+Now that we have the definitions file, we can build the singularity image with:
+
+.. code-block:: bash
+   
+   singularity build --fakeroot --nv --sandbox cuda_example.sif cuda_example.def
+
+In this command some flags are used, these and more are explained in the table below.
+
+===============   ======================================================================================
+Flag              Functionality         
+===============   ======================================================================================
+``--fakeroot``    raises permissions inside the container to ``sudo``, necessary for installing packages
+``--nv``          exposes the nvidia drivers of the host to the container (makes them available)
+``--sandbox``     allows the final container to be changed in *write-mode*, should only be used for debugging!
+``--writable``    allows writing into a sandboxed container when invoking ``singularity shell``
+===============   ======================================================================================
+
+``--fakeroot`` is needed for installing ``git`` and ``make`` in the container. ``--nv`` is necessary to access the GPU from within the container, and ``--sandbox`` is used to allow the user after running this example to go into the container and make changes to folders, files or run other commands that change the state of the container.
+
+Once the container is built - which can take a few minutes as multiple base containers have to be retrieved from the internet - you can run it using 
+
+.. code-block:: bash
+
+   singularity run --nv cuda_example.sif
+
+which will output the result of the *eigenvalues-test*, as was instructed in the definitions file under ``%runscript``. To run commands from within a shell in the container that allow for making changes, do
+
+.. code-block:: bash
+
+   singularity shell --nv --writable gpu_test.sif
```

Comment:
For consistency perhaps use the same image in the example:
``singularity shell --nv --writable cuda_example.sif``